lib/graphhopper/api_client.rb

```diff
@@ -277,6 +281,11 @@ def build_request_body(header_params, form_params, body)
       else
         data = nil
       end
+
+      if data && header_params['Content-Encoding'] == 'gzip'
```

Comment:
Hmm... I think that after introducing `use_compression` we can change `header_params['Content-Encoding'] == 'gzip'` to `use_compression`.

Reply:
🙈